src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -0,0 +1,197 @@
+package edu.berkeley.ce.rockslicing
+
+import breeze.linalg
+import breeze.linalg.{DenseVector, DenseMatrix}
+
+object BlockVTK {
+
+  private val EPSILON = 1.0e-6
+
+  /**
+   * Calculates the rotation matrix to rotate the input plane (specified by its normal)
+   * to the desired orientation (specified by the desired normal)
+   * @param n_current: Current normal of plane
+   * @param n_desired: Desired new normal
+   * @return 3*3 rotation matrix
+   */
+  private def rotationMatrix(n_current: (Double, Double, Double), n_desired: (Double, Double, Double)):
+  DenseMatrix[Double] = {
+    val n_c = DenseVector[Double](n_current._1, n_current._2, n_current._3)
+    val n_d = DenseVector[Double](n_desired._1, n_desired._2, n_desired._3)
+    if (math.abs(linalg.norm(linalg.cross(n_c,n_d))) > BlockVTK.EPSILON) {
+      val (u, v, w) = n_current
+
+      // Rotation matrix to rotate into x-z plane
+      val Txz = DenseMatrix.zeros[Double](3, 3)
+      Txz(0,0) = u / math.sqrt(u*u + v*v)
+      Txz(1,0) = -v /math.sqrt(u*u + v*v)
+      Txz(0,1) = v / math.sqrt(u*u + v*v)
+      Txz(1,1) = u / math.sqrt(u*u + v*v)
+      Txz(2,2) = 1.0
+
+      // Rotation matrix to rotate from x-z plane on z-axis
+      val Tz = DenseMatrix.zeros[Double](3, 3)
+      Tz(0,0) = w / math.sqrt(u*u + v*v + w*w)
+      Tz(2,0) = math.sqrt(u*u + v*v) / math.sqrt(u*u + v*v + w*w)
+      Tz(0,2) = -math.sqrt(u*u + v*v)/math.sqrt(u*u + v*v + w*w)
+      Tz(2,2) = w / math.sqrt(u*u + v*v + w*w)
+      Tz(1,1) = 1.0
+      Tz * Txz
+    } else {
+      DenseMatrix.eye[Double](3)
+    }
+  }
+
+    /**
+   * Finds the center of a list of vertices - defined as the average coordinate of all the vertices in the list
+   * @param vertices List of vertices
+   * @return Coordinate of center of vertices
+   */
+  private def findCenter(vertices: Seq[(Double, Double, Double)]): (Double, Double, Double) = {
+    (vertices.map(_._1).sum / vertices.length.toDouble,
+      vertices.map(_._2).sum / vertices.length.toDouble,
+      vertices.map(_._3).sum / vertices.length.toDouble)
+  }
+
+  /**
+   * Compares pointA to pointB. If pointA is first relative to pointB rotating counter-clockwise about center,
+   * the method returns true. Input vectors should have same z-coordinate - comparison is based in 2-D
+   * @param pointA Point of interest
+   * @param pointB Point of comparison
+   * @param center Reference point for comparison
+   * @return Returns TRUE if pointA is first relative to pointB, FALSE otherwise
+   */
+  private def ccwCompare(pointA: (Double, Double, Double), pointB: (Double, Double, Double),
+                           center: (Double, Double, Double)): Boolean = {
```

Comment:
See, I thought I was doing this... Seems like I'm a bit messy, will try to be a bit more hygienic.